include/oneapi/dpl/pstl/glue_memory_impl.h

```diff
@@ -141,22 +141,22 @@ template <class _ExecutionPolicy, class _ForwardIterator, class _Tp>
 oneapi::dpl::__internal::__enable_if_execution_policy<_ExecutionPolicy>
 uninitialized_fill(_ExecutionPolicy&& __exec, _ForwardIterator __first, _ForwardIterator __last, const _Tp& __value)
 {
-    typedef typename ::std::iterator_traits<_ForwardIterator>::value_type _ValueType;
-    typedef ::std::decay_t<_ExecutionPolicy> _DecayedExecutionPolicy;
+    typedef typename std::iterator_traits<_ForwardIterator>::value_type _ValueType;
+    typedef std::decay_t<_ExecutionPolicy> _DecayedExecutionPolicy;
 
     auto __dispatch_tag = oneapi::dpl::__internal::__select_backend(__exec, __first);
 
-    if constexpr (::std::is_arithmetic_v<_ValueType>)
+    if constexpr (std::is_trivially_default_constructible_v<_ValueType> && std::is_trivially_copyable_v<_ValueType>)
```

Comment:
We should check `is_trivially_assignable` as well, if we want to keep this branch. Otherwise, there is no guarantee that the operation is available to call (`is_trivially_copyable` is not sufficient by itself).